### PR TITLE
add compat data for Critical-CH

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

This PR provides compat-data for the `Critical-CH` HTTP header (see https://chromestatus.com/feature/5727177800679424 for more details).

Note: I have had to set this to `"standard_track": false` and remove the `spec_url` field, because the linter wouldn't pass it with any of the available specifications for the feature. The spec URLs I found were all archive or experimental versions, e.g. https://www.ietf.org/archive/id/draft-davidben-http-client-hint-reliability-03.txt. Any suggestions would be much appreciated.

### CODE REVIEW

**Comment by Elchi3** [comment]
> Note: I have had to set this to `"standard_track": false` and remove the `spec_url` field, because the linter wouldn't pass it with any of the available specifications for the feature. The spec URLs I found were all archive or experimental versions, e.g. https://www.ietf.org/archive/id/draft-davidben-http-client-hint-reliability-03.txt. Any suggestions would be much appreciated.

We're using https://github.com/w3c/browser-specs to tell what is a real specification. 
We also have a few exceptions: https://github.com/mdn/browser-compat-data/blob/main/test/linter/test-spec-urls.ts#L15 It is possible to add exceptions so that our linter passes, but usually we first file an issue with https://github.com/w3c/browser-specs to ask why a particular spec isn't included (yet).

File: http/headers/Critical-CH.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            "safari": {
+              "version_added": false
+            },
+            "safari_ios": "mirror",
+            "samsunginternet_android": "mirror",
+            "webview_android": "mirror"
+          },
+          "status": {
+            "experimental": true,
+            "standard_track": false,
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply http/headers/Critical-CH.json:33]: It says "Expired Internet-Draft" so not sure if we should consider it as standard track. We rely on https://github.com/w3c/browser-specs to know which spec urls are relevant.

**Comment by Elchi3** [comment]
Let's wait a day or two. Usually Dom and/or Francois are quick over on w3c/browser-specs :) 
Otherwise you will need to update this PR to add the spec to BCD's exception list: https://github.com/mdn/browser-compat-data/blob/main/test/linter/test-spec-urls.ts#L15

**Comment by Elchi3** [comment]
Yes sir! Just a few more hours of patience then the new package version will be merged into BCD main and you can rebase this PR to it. Feel free to add the spec_url and flip the standard_track status.

**Comment by Elchi3** [comment]
https://github.com/mdn/browser-compat-data/pull/18423 has been merged.

File: http/headers/Critical-CH.json (hunk 1 above)
Comment (Elchi3) [inline http/headers/Critical-CH.json:33]:
```suggestion
            "standard_track": true,
```

File: http/headers/Critical-CH.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+  "http": {
+    "headers": {
+      "Critical-CH": {
+        "__compat": {
+          "description": "<code>Critical-CH</code> request header",
+          "mdn_url": "https://developer.mozilla.org/docs/Web/HTTP/Headers/Critical-CH",
```
Comment (Elchi3) [inline http/headers/Critical-CH.json:7]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/HTTP/Headers/Critical-CH",
          "spec_url": "https://datatracker.ietf.org/doc/html/draft-davidben-http-client-hint-reliability",
```

**Review by Elchi3 — CHANGES_REQUESTED** [review]

**Comment by chrisdavidmills** [comment]
Awesome, cheers @Elchi3 !

File: http/headers/Critical-CH.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{
+  "http": {
+    "headers": {
+      "Critical-CH": {
+        "__compat": {
+          "description": "<code>Critical-CH</code> request header",
```
Comment (Elchi3) [inline http/headers/Critical-CH.json:6]:
```suggestion
```

**Review by Elchi3 — APPROVED** [review]

**Comment by chrisdavidmills** [comment]
@Elchi3 nice one, thanks mate

**Pull request merged**